Wrap up functors

devesh-shetty opened the pull request:
This PR wraps up the functor doc

devesh-shetty committed:
wrap up functors (close #9)

i-walker commented on src/main/ank/1.7-functors-Part1.md:
````diff
@@ -52,7 +57,7 @@ identity(x) == x
 ```
 ```kotlin
-fmap { s: A -> identity(s) } == identity
+fmap(f = ::identity) == ::identity
````
This is more readable 
```suggestion
fmap { s: A -> identity(s) } == identity
```

i-walker commented on src/main/ank/1.7-functors-Part1.md:
````diff
@@ -63,7 +68,10 @@ Nothing
 ```
 ```markdown
-fmap {identity(it)} (None)
+import arrow.core.None
````
These imports are unnecessary because this is not compilable code.

i-walker commented on src/main/ank/1.7-functors-Part1.md:
```diff
@@ -63,7 +68,10 @@ Nothing
+import arrow.core.None
+import arrow.core.identity
+
+fmap(f = ::identity, o = None)
```
```suggestion
fmap { identity(it) } (None)
```

devesh-shetty replied:
🤔The method signature of `fmap` has changed in the doc
`fun <A, B> fmap(f: (A) -> B, o: Option<A>): Option<B>`
Won't that make `fmap { identity(it) } (None)` invalid?

i-walker replied:
which doc are you always referring to? 
The aim of this project is to translate the book.

i-walker replied:
If you referring to the Arrow-docs. The nearest what we try to model is this [function](https://github.com/arrow-kt/arrow/blob/ffd5409f6ce0594d61149d4c68964f27b651bb2a/modules/core/arrow-core-data/src/main/kotlin/arrow/core/Option.kt#L81) here.

i-walker commented on src/main/ank/1.7-functors-Part1.md:
````diff
@@ -80,7 +88,10 @@ Just x
 ```
 ```markdown
-fmap {identity(it)} (Some(x))
+import arrow.core.Some
````
Same

i-walker commented on src/main/ank/1.7-functors-Part1.md:
```diff
@@ -80,7 +88,10 @@ Just x
+import arrow.core.Some
+import arrow.core.identity
+
+fmap(f = ::identity, o = Some(x))
```
```suggestion
fmap { identity(it) } (Some(x))
```

i-walker commented on src/main/ank/1.7-functors-Part1.md:
````diff
@@ -110,17 +121,17 @@ fmap(g compose f) == fmap(g) compose fmap(f)
 ```
 ```markdown
-fmap(g compose f) (Some(x))
+fmap(f = g compose f, o = Some(x))
````
```suggestion
fmap(g compose f) (Some(x))
```

i-walker commented on src/main/ank/1.7-functors-Part1.md:
```diff
@@ -110,17 +121,17 @@ fmap(g compose f) == fmap(g) compose fmap(f)
 Some(g(f(x)))
 = { definition of fmap }
-fmap(g)(Some(f(x)))
+fmap(f = g, o = Some(f(x)))
```
```suggestion
fmap(g)(Some(f(x)))
```

i-walker commented on src/main/ank/1.7-functors-Part1.md:
```diff
@@ -165,11 +176,10 @@ public:
 
-    fun isDefined(): Boolean = !isEmpty() // Some
-    //…
+    fun isDefined(): Boolean = !isEmpty()
```
It's just agnostic to the book code
```suggestion
    fun isDefined(): Boolean = !isEmpty() // Some
```

i-walker commented:
@devesh-shetty Thanks for cleaning up 👍🏾. Now finish the remaining 8 - 9 pages of the chapter and we can review it.

i-walker commented on src/main/ank/1.7-functors-Part1.md:
```diff
@@ -121,13 +115,13 @@ fmap(g compose f) == fmap(g) compose fmap(f)
-fmap(f = g, o = Some(f(x)))
+fmap(g)(Some(f(x)))
 = { definition of fmap }
 fmap(f = g, o = fmap(f = f, o = Some(x)))
```
same here
```suggestion
fmap(g)(fmap(f)(Some(x)))
```

i-walker commented on src/main/ank/1.7-functors-Part1.md:
```diff
@@ -118,9 +123,9 @@ Some(g(f(x)))
+fmap(f = g, o = fmap(f = f, o = Some(x)))
 = { definition of compose }
-(fmap(g) compose fmap(f))(Some(x))
+(fmap(f = g) compose fmap(f = f))(Some(x))
```
```suggestion
(fmap(g) compose fmap(f))(Some(x))
```

i-walker commented on src/main/ank/1.7-functors-Part1.md:
````diff
@@ -34,11 +39,11 @@ fun <A, B> fmap(f: (A) -> B): (Option<A>) -> Option<B> =
 fmap :: (a -> b) -> Maybe a -> Maybe b
 ```
-```kotlin:ank:silent
+```kotlin:ank:playground
````
```suggestion
```kotlin:ank:silent
```
Ask if you don't know what the tags mean. 
Silent tag needs to be executed, because we will use it later on in a subsequent snippet.
Check out the ReadMe file. I updated it.

i-walker commented:
Hey, @devesh-shetty thank you for your contribution. I mentioned you in the last commit in master - `clean-up was heavily inspired by @devesh-shetty PR#50`. 
I could not get a hang of you :) mad I did not find you in other Arrow related communities. 

I hope you're not mad. Just look for another open Issue. I will finish this one. 
You may close this issue or continue, if 1.7 is not finished, yet.

devesh-shetty commented:
> Hey, @devesh-shetty thank you for your contribution. I mentioned you in the last commit in master - `clean-up was heavily inspired by @devesh-shetty PR#50`.
> I could not get a hang of you :) mad I did not find you in other Arrow related communities.

Hey, apologies, I have rebased and addressed the comments, could you please take another look?
I am in the Arrow slack channel, are they any other communities to join?

> I hope you're not mad. Just look for another open Issue. I will finish this one.
> You may close this issue or continue, if 1.7 is not finished, yet.

Naah, apologies on making you wait thing long 😅

i-walker commented:
This is probably not the way to accomplish the issue, which is related to this PR.

I kindly ask you to read through the Readme file of this Project and start a new PR on 1.8, which is still open. First come, first served ;)
